doc/reference.rst

```diff
@@ -42,3 +53,215 @@ It then runs the resulting buildout.
+  the configuration file that defined the value. (If it contains value
+  substitutions, and the result is a relative path, then it will be
+  interpreted relative to the buildout directory.)
+
+eggs-directory, default: 'eggs'
+  The directory where :ref:`eggs <eggs-label>` are installed.
+
+  This is often set in in a :ref:`User-default configuration
+  <user-default-configuration>` to share eggs between buildouts.
+  See the section on :doc:`Optimizing buildouts with shared eggs and
+  download caches <topics/optimizing>`.
+
+  If the value is a relative path and doesn't contain value
+  substitutions, it's interpreted relative to the directory containing
+  the configuration file that defined the value. (If it contains value
+  substitutions, and the result is a relative path, then it will be
+  interpreted relative to the buildout directory.)
+
+executable, default: sys.executable, read-only
+  The full path to the Python executable used to run the buildout.
+
+.. _extends-option-ref:
+
+extends
+  The names, separated by whitespace, of one or more configurations
+  that the configuration containing the ``extends`` option should
+  :ref:`extend <extends_option>`. The names may be file paths, or
+  URLs.  If they are relative paths, they are interpreted relative to
+  the configuration containing the ``extends`` option.
+
+.. _extends-cache-buildout-option:
+
+
+extends-cache
+  An optional directory to cache remote configurations in.  This
+  allows remote configurations to be used in :ref:`offline mode
+  <offline-mode>`.  When not in offline mode, and a remote
+  configuration is used, by using a URL in an
+  :ref:`extends option <extends_option>`, the configuration is written
+  to the extends cache and will be used if the buildout is later run
+  in offline mode.
+
+  If the value is a relative path and doesn't contain value
+  substitutions, it's interpreted relative to the directory containing
+  the configuration file that defined the value. (If it contains value
+  substitutions, and the result is a relative path, then it will be
+  interpreted relative to the buildout directory.)
+
+find-links, default: ''
+  Extra locations to search for distributions to download.
+
+  These may be file paths or URLs.  These may name individual
+  distributions or directories containing
+  distributions. Subdirectories aren't searched.
+
+index
+  An alternate index location.
+
+  This can be a local directory name or an URL.  It can be a flat
+  collection of distributions, but should be a "simple" index, with
+  subdirectories for distribution `project names
+  <https://packaging.python.org/distributing/#name>`_ containing
+  distributions for those projects.
+
+  If this isn't set, then ``https://pypi.python.org/simple/`` is used.
+
+install-from-cache, default: 'false'
+  If this is true, then distributions will only be installed if they
+  can be found in the :ref:`download cache <download-cache>`.
```

Comment:
And if a distribution is not found in the cache, buildout will exit with an error message?

Maybe worth mentioning why you'd want to set this option.  If I understand correctly, it's when you want to make sure offline installs are possible?